shell/creators/extension/init

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env node
+/* eslint-disable no-console */
+
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const args = process.argv.slice(2);
+let extensionName = '';
+let appName = '';
+let updateOnly = false;
+let skeletonOnly = false;
+
+args.forEach((arg, index) => {
+  switch (arg) {
+  case '--update':
+  case '-u':
+    updateOnly = true;
+    break;
+  case '--app-name':
+  case '-a':
+    if ( args[index + 1] && !args[index + 1].startsWith('-') ) {
+      appName = args[index + 1];
+    } else {
+      console.error('Error: Missing value for --app-name or -a option.');
+      process.exit(1);
+    }
+    break;
+  case '--skeleton-only':
+  case '-s':
+    skeletonOnly = true;
+    break;
+  default:
+    if ( !arg.startsWith('-') && extensionName === '' ) {
```

Comment:
We do need it, when you remove that condition the code no longer checks if `extensionName` is already set before assigning it. So the first non-option argument will always be assigned to `extensionName`, when there are multiple arguments the `extensionName` will be overwritten with the subsequent arguments.

For instance if you create an extension with:

```sh
yarn create @rancher/extension my-pkg -a my-skeleton
```

^ The `extensionName` will first be assigned `my-pkg`, then after the `-a` option is passed, it will be assigned `my-skeleton`.